backend/rest_api_service/main.py

```diff
@@ -62,3 +64,38 @@ async def get_all_participants(response: Response, db: Session = Depends(get_db)
     except Exception as e:
         response.status_code = 500
         return {"message": f"An error occured: {e}"}
+    
+
+@app.get("/login/{username}", tags=['User'], status_code=status.HTTP_200_OK)
+async def login(username: str, password: str, response: Response, db: Session = Depends(get_db)):
+    try: 
+        user = await crud.get_user_by_username(db, username)
+
+        if(bcrypt.checkpw(password.encode('utf-8'), user.password_hash.encode('utf-8'))):
+            await crud.update_user_last_login(db, user.user_id)
+            return user
+        else:
+            raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Incorrect password")
+    except Exception as e:
+        response.status_code = 500
+        return {"message": f"An error occured: {e}"}
+    
+
+@app.post("/register", tags=['User'], status_code=status.HTTP_201_CREATED)
+async def register(username: str, password: str, email : str, birthday : datetime.datetime, response: Response, db: Session = Depends(get_db)):
```

Comment:
Tutaj tak samo jak z logowaniem. Plus jeszcze trzeba dodać walidację emaila, możesz to tego użyć klasy EmailStr z pydantic (https://fastapi.tiangolo.com/uk/tutorial/extra-models/)